analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SecurityPInvokeMethodShouldNotBeCalled.vb

```diff
@@ -0,0 +1,61 @@
+﻿Imports System
+Imports System.Runtime.InteropServices
+
+Namespace Tests.TestCases
+
+     Public Class Noncompliant
+
+    <DllImport("ole32.dll")>
+    Public Shared Function CoSetProxyBlanket(<MarshalAs(UnmanagedType.IUnknown)>pProxy As Object, dwAuthnSvc as UInt32, dwAuthzSvc As UInt32, <MarshalAs(UnmanagedType.LPWStr)> pServerPrincName As String, dwAuthnLevel As UInt32, dwImpLevel As UInt32, pAuthInfo As IntPtr, dwCapabilities As UInt32) As Integer
+    End Function
+
+    Public Enum RpcAuthnLevel
+        [Default] = 0
+        None = 1
+        Connect = 2
+        [Call] = 3
+        Pkt = 4
+        PktIntegrity = 5
+        PktPrivacy = 6
+    End Enum
+
+    Public Enum RpcImpLevel
+        [Default] = 0
+        Anonymous = 1
+        Identify = 2
+        Impersonate = 3
+        [Delegate] = 4
+    End Enum
+
+    Public Enum EoAuthnCap
+        None = &H00
+        MutualAuth = &H01
+        StaticCloaking = &H20
+        DynamicCloaking = &H40
+        AnyAuthority = &H80
+        MakeFullSIC = &H100
+        [Default] = &H800
+        SecureRefs = &H02
+        AccessControl = &H04
+        AppID = &H08
+        Dynamic = &H10
+        RequireFullSIC = &H200
+        AutoImpersonate = &H400
+        NoCustomMarshal = &H2000
+        DisableAAA = &H1000
+    End Enum
+
+    <DllImport("ole32.dll")>
+    Public Shared Function CoInitializeSecurity(pVoid As IntPtr, cAuthSvc As Integer, asAuthSvc As IntPtr, pReserved1 As IntPtr, level As RpcAuthnLevel, impers As RpcImpLevel, pAuthList As IntPtr, dwCapabilities As EoAuthnCap, pReserved3 As IntPtr) As Integer
+    End Function
+
```

Comment:
VB.NET has also alternative (and preferred in VB world) syntax to do this that we should also cover:
```
    Declare Function CoInitializeSecurity Lib "ole32.dll" (pVoid As IntPtr, cAuthSvc As Integer, asAuthSvc As IntPtr, pReserved1 As IntPtr, level As RpcAuthnLevel, impers As RpcImpLevel, pAuthList As IntPtr, dwCapabilities As EoAuthnCap, pReserved3 As IntPtr) As Integer
```
and other variant of the same so we need to check for the identifier in Alias
```
    Declare Function CustomName Lib "ole32.dll" Alias "CoInitializeSecurity" (pVoid As IntPtr, cAuthSvc As Integer, asAuthSvc As IntPtr, pReserved1 As IntPtr, level As RpcAuthnLevel, impers As RpcImpLevel, pAuthList As IntPtr, dwCapabilities As EoAuthnCap, pReserved3 As IntPtr) As Integer
```
The alias has a priority, so let's document in UT that we don't have FP on this:
```
    Declare Function CoInitializeSecurity Lib "ole32.dll" Alias "SomethingElse" (pVoid As IntPtr, cAuthSvc As Integer, asAuthSvc As IntPtr, pReserved1 As IntPtr, level As RpcAuthnLevel, impers As RpcImpLevel, pAuthList As IntPtr, dwCapabilities As EoAuthnCap, pReserved3 As IntPtr) As Integer

Reply:
I'm not sure how much work it would be to implement the `Alias` correctly and how much can you get from semantic model. 

Let's say `CoInitializeSecurity Lib "ole32.dll" Alias "SomethingElse"` is acceptable FP (let's just document it if we can't handle it). Why would you do it anyway?

I'm worried about `CustomName Lib "ole32.dll" Alias "CoInitializeSecurity"` as current implementation filters the `CustomName` out early as this makes sense for C#.